File: meetupCpp/sort.cpp

#include <vector>
#include <string>
#include <fstream>
#include <iostream>
#include <chrono>
#include <algorithm>

using namespace std;

vector<string> lines_from_file(string filename){
    vector<string> lines;

    ifstream file(filename.c_str());
    string line;
    while(getline(file, line)){
        lines.push_back(line);
    }

    return std::move(lines);
}


int main(){
    auto lines = lines_from_file("/home/alexey/tmp/data.bin");

    cout<<"Read: "<< lines.size() <<" lines"<<endl;

    auto begin = chrono::steady_clock::now();
    sort(lines.begin(), lines.end());
    auto end = chrono::steady_clock::now();

    auto elapsed_ms = chrono::duration_cast<chrono::milliseconds>(end - begin);
    cout << "The time: " << elapsed_ms.count() << " ms\n";

    return 0;
}

File: meetupRust/src/sort.rs
use std::time::Instant;

use std::{
    fs::File,
    io::{prelude::*, BufReader},
    path::Path,
};

fn lines_from_file(filename: impl AsRef<Path>) -> Vec<String> {
    let file = File::open(filename).expect("no such file");
    let buf = BufReader::new(file);
    buf.lines()
        .map(|l| l.expect("Could not parse line"))
        .collect()
}

fn main() {
    let mut lines = lines_from_file("/home/alexey/tmp/data.bin");

    println!("Read: {} lines", lines.len());

    let now = Instant::now();
    lines.sort();
    let elapsed = now.elapsed().as_millis();

    println!("Sorted: {} ms", elapsed);

}